include/dylib.hpp

```diff
@@ -215,7 +215,7 @@ class dylib {
     }
 
 protected:
-    native_handle_type m_handle{nullptr};
+    native_handle_type m_handle{};
```

Comment:
Now you're making two assumptions: constructible from 0 and default constructible :-P